src/internal/document_api/mod.rs

```diff
@@ -522,28 +708,148 @@ fn encrypt_document_core<'a, CR: rand::CryptoRng + rand::RngCore>(
                 dedupe_grants(&grants),
             );
 
-            (
-                grants,
-                encrypted_doc,
-                vec![
-                    accum_errs,
-                    encrypt_errs.into_iter().map(|e| e.into()).collect(),
-                ]
-                .into_iter()
-                .concat(),
-            )
+            EncryptionResult {
+                edeks: grants
+                    .into_iter()
+                    .map(|(wk, ev)| EncryptedDek {
+                        grant_to: wk,
+                        encrypted_dek_data: ev,
+                    })
+                    .collect(),
+                encrypted_data: encrypted_doc,
+                encryption_errs: vec![encrypt_errs.into_iter().map(|e| e.into()).collect()]
+                    .into_iter()
+                    .concat(),
+            }
         })
     }
-    .into_future()
-    .and_then(move |(grants, encrypted_doc, other_errs)| {
-        document_create::document_create_request(auth, doc_id.clone(), document_name, grants)
-            .map(move |resp| (doc_id, resp, encrypted_doc, other_errs))
+}
+
+/// An encrypted document encryption key.
+///
+/// Once decrypted, the DEK serves as a symmetric encryption key.
+///
+/// It can also be useful to think of an EDEK as representing a "document access grant" to a user/group.
+#[derive(Debug, Clone, PartialEq)]
+pub struct EncryptedDek {
+    grant_to: WithKey<UserOrGroup>,
+    encrypted_dek_data: recrypt::api::EncryptedValue,
+}
+
+impl From<&EncryptedDek> for EncryptedDekP {
+    fn from(edek: &EncryptedDek) -> Self {
+        use crate::proto::edeks;
+        use recrypt::api as re;
+        let edek = edek.clone();
+
+        // transform the recrypt EncryptedValue to a edek proto
```

Comment:
`transform` is a little confusing - I was thinking transform crypto when I saw it. Maybe `encode the EncryptedValue as a protobuf`?